app/gui2/src/composables/nodeCreation.ts

```diff
@@ -173,8 +173,19 @@ export function useNodeCreation(
     }
   }
 
-  function createNode(options: NodeCreationOptions) {
-    createNodes([options])
+  let delayedNodesToCreate: NodeCreationOptions[] = []
+
+  function scheduleCreateNode(options: NodeCreationOptions) {
+    delayedNodesToCreate.push(options)
+    // Delay node creation to next tick, batch multiple synchronous createNode calls together
+    // to avoid node name collisions.
+    if (delayedNodesToCreate.length === 1) {
```

Comment:
A strict `=== 1` comparison seems suspicious, why not `>= 1`?

Reply:
It schedules a flush operation when the queue transitions from empty to non-empty. If the queue is already non-empty (i.e. after adding an element, the queue's new length is 2 or more), a flush has already been scheduled.